src/encryptionkey.rs

```diff
@@ -68,51 +72,92 @@ impl EncryptionKey {
     where
         M: AsRef<[u8]>,
     {
-        let xx = BigNumber::from_slice(x);
-        if !mod_in(&xx, &self.n) {
+        let m = BigNumber::from_slice(x);
+        let r = r.unwrap_or_else(|| Nonce::random(&self.n));
+
+        if !mod_in(&m, &self.n) {
+            println!("m");
```

Comment:
delete this `println!`

Reply:
Ah, not sure why I thought the old code had it, probably from my debugging session.